src/main/java/ch/njol/skript/expressions/ExprEntities.java

```diff
@@ -180,35 +181,16 @@ public Iterator<? extends Entity> iterator(Event e) {
 					return false;
 				});
 		} else {
-			if (worlds == null && returnType == Player.class)
+			if (worlds == null && chunks == null && returnType == Player.class)
 				return super.iterator(e);
 
-			return new NonNullIterator<Entity>() {
-					private World[] ws = worlds == null ? Bukkit.getWorlds().toArray(new World[0]) : worlds.getArray(e);
-					private EntityData<?>[] ts = types.getAll(e);
-					private int w = -1;
-					@Nullable
-					private Iterator<? extends Entity> curIter = null;
-
-					@Override
-					@Nullable
-					protected Entity getNext() {
-						while (true) {
-							while (curIter == null || !curIter.hasNext()) {
-								w++;
-								if (w == ws.length)
-									return null;
-								curIter = ws[w].getEntitiesByClass(returnType).iterator();
-							}
-							while (curIter.hasNext()) {
-								Entity current = curIter.next();
-								for (EntityData<?> t : ts) {
-									if (t.isInstance(current))
-										return current;
-								}
-							}
-						}
-					}};
+			Entity[] entities;
+			if (chunks != null) {
+				entities = EntityData.getAll(types.getArray(e), returnType, chunks.getArray(e));
+			} else {
+				return super.iterator(e);
+			}
+			return Arrays.stream(entities).iterator();
```

Comment:
The world isn't even used, so do we need the world null check? Also you're doing two chunk null checks, can't this be simplified to
```java
if (chunks == null || returnType == Player.class)
	return super.iterator(e);
return Arrays.stream(EntityData.getAll(types.getArray(e), returnType, chunks.getArray(e))).iterator();
```

Reply:
Nice catch! I have done some more tests and all seem to work
![image](https://user-images.githubusercontent.com/20037329/176599120-52b14f8a-5179-4df6-a931-fe720b58c4de.png)